src/AppBundle/Controller/AuthenticationController.php

```diff
@@ -29,74 +30,127 @@ class AuthenticationController extends Controller
+        $logContext = ['nameId' => $nameId];
+
 
         // Do have a valid sample AuthnRequest?.
         if (!$this->authenticationService->authenticationRequired()) {
+            $this->logger->error(
+                'there is no pending authentication request from SP',
+                $logContext
+            );
+
             return new Response('No authentication required', Response::HTTP_BAD_REQUEST);
         }
 
+
         // Handle one time password
         if ($request->get('otp') !== null) {
+            $this->logger->info('handling otp', $logContext);
+
             // Implement otp
-            return new Response('One time passord: '. $request->get('otp'), Response::HTTP_I_AM_A_TEAPOT);
+            return new Response('One time passord: '.$request->get('otp'), Response::HTTP_I_AM_A_TEAPOT);
         }
 
+        $this->logger->info('Verify if the user is already authenticated', $logContext);
+
         // Are we already logged in with tiqr?
         if ($this->tiqrService->isAuthenticated()) {
+            $this->logger->info('Authentication is finalized, returning to SP', $logContext);
             $this->authenticationService->authenticate();
 
             return $this->authenticationService->replyToServiceProvider();
         }
 
         // Start authentication process.
         try {
+            $this->logger->info('start authentication', $logContext);
             $this->tiqrService->startAuthentication($nameId);
         } catch (\Exception $e) {
+            $this->logger->error(sprintf(
+                'Failed to start authentication "%s"',
+                $e->getMessage()
+            ), $logContext);
+            $this->logger->info(
+                'Returning authentication failed response',
+                $logContext
+            );
             $this->authenticationService->reject($e->getMessage());
 
             return $this->authenticationService->replyToServiceProvider();
         }
 
+        $this->logger->info(
+            'Returning authentication page with QR code',
+            $logContext
+        );
+
         return $this->render('AppBundle:default:authentication.html.twig', []);
     }
 
     /**
      * @Route("/authentication/status", name="app_identity_authentication_status")
+     * @throws \InvalidArgumentException
      */
     public function authenticationStatusAction()
     {
+        $this->logger->info('Request for authentication status');
+
         if (!$this->authenticationService->authenticationRequired()) {
+            $this->logger->error('there is no pending authentication request from SP');
+
             return new Response('No authentication required', Response::HTTP_BAD_REQUEST);
         }
 
-        return new JsonResponse($this->tiqrService->isAuthenticated());
+        $isAuthenticated = $this->tiqrService->isAuthenticated();
+
+        if ($isAuthenticated) {
+            $this->logger->info('Send json response is authenticated');
```

Comment:
You mean sent, nod send.

Reply:
A message can be authenticated, but that is not what you mean. Proposal: Authentication successful, sending JSON response.